tee-worker/litentry/core/vc-task/receiver/src/lib.rs

```diff
@@ -84,62 +91,300 @@ pub fn run_vc_handler_runner<ShieldingKeyRepository, A, S, H, O, Z, N>(
 	let vc_task_receiver = init_vc_task_sender_storage();
 	let pool = ThreadPool::new().unwrap();
 
-	let (sender, receiver) = channel::<(ShardIdentifier, TrustedCall)>();
+	let (tc_sender, tc_receiver) = channel::<(ShardIdentifier, TrustedCall)>();
 
 	// Spawn thread to handle received tasks, to serialize the nonce increase even if multiple threads
 	// are submitting trusted calls simultaneously
 	let context_cloned = context.clone();
 	thread::spawn(move || loop {
-		if let Ok((shard, call)) = receiver.recv() {
+		if let Ok((shard, call)) = tc_receiver.recv() {
 			info!("Submitting trusted call to the pool");
 			if let Err(e) = context_cloned.submit_trusted_call(&shard, None, &call) {
 				error!("Submit Trusted Call failed: {:?}", e);
 			}
 		}
 	});
 
+	// use local registry to manage request reponse status
+	let req_registry = RequestRegistry::new();
+	let req_registry_cloned = req_registry.clone();
+	thread::spawn(move || loop {
+		// Doesn't harm anything to clean up slowly.
+		let wait_seconds = 200u64;
+		req_registry_cloned.clean_up(wait_seconds);
+		sleep(Duration::from_secs(wait_seconds));
+	});
```

Comment:
It sounds like something could be done via reference counting - like `Arc` or so - not in this PR though...